Program.cs

```diff
@@ -101,29 +102,59 @@ private static void WriteToDss(string interval, DateTime t, DssWriter dss, int s
         parameter = "Temperature";
       }
       string dssPath = "/Trinity/" + dsn + "/" + parameter + "//" + interval + "/" + F + "/";
-      WriteToDss(dss, data, dssPath, t, units, dataType);     
+      WriteToDss(dss, data, dssPath, t, units, dataType);
     }
 
-    private static void WriteToDss(DssWriter dss, float[] data, string dssPath,DateTime startTime, string units, string dataType)
+    private static void WriteToDss(DssWriter dss, float[] data, string dssPath, DateTime startTime, string units, string dataType)
     {
-      double[] d = new double[data.Length];
-      Array.Copy(data, d, d.Length);
-      // insert leap days... copy prev value...
+      double[] d = addLeapDays(data, startTime);
       Hec.Dss.TimeSeries ts = new TimeSeries(dssPath, d, startTime, units, dataType);
-      
-      try{
+
+      try
+      {
         dss.Write(ts, true);
-      }catch(AccessViolationException e){
+      }
+      catch (AccessViolationException e)
+      {
         Console.WriteLine("Access Violation Occurred, trying again. " + e.Message);
-        try{
+        try
+        {
           dss.Write(ts, true);
-        }catch(AccessViolationException e2){
-          File.WriteAllText("WriteExceptions.txt","Exception " + e2.Message + " " + dssPath);
         }
-      }catch(Exception ex){
+        catch (AccessViolationException e2)
+        {
+          File.WriteAllText("WriteExceptions.txt", "Exception " + e2.Message + " " + dssPath);
+        }
+      }
+      catch (Exception ex)
+      {
         Console.WriteLine("Exception " + ex.Message + " " + dssPath);
       }
     }
 
+    private static double[] addLeapDays(float[] inputdata, DateTime inputT)
+    {
+      int totalCount = inputdata.Length;
+      List<double> output = new List<double>(totalCount + 24 * inputdata.Length / 24 / 364);
```

Comment:
does this add a day for every year? we only need to add it for leap years, which is less?
If we dont care about the efficency of the program we can default the list to totat count and then add values when necessary at the end - lists can dynamically resize.

Reply:
The constructor is for the capacity, while the initial size is zero. The code has a bug that we don't ever Add to the list.
We could make capacity a little bigger to prevent allocations.

Reply:
how does .ToArray() work with an allocation larger than the data size?

Reply:
It may be a copy.  https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.toarray?view=netframework-4.7.2

Reply:
i think your proposal to switch to add for all is the best idea